### PR TITLE
Listing expiration & fees

### PR DESCRIPTION
**Pull request by AllienWorks**
screenshot from 2018-07-17 16-35-43

Added expiration input in `Add listing` page with estimated fee.

- [ ] needs implementation of fee estimates (@xludx has some pointers regarding this)

### CODE REVIEW

**Comment by vikas-cis** [comment]
@AllienWorks can you please have a look some small css is required after changes :)

**Comment by AllienWorks** [comment]
@kewde can we merge this?

File: src/app/core/market/api/listing/listing.service.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export class ListingService {
     return this.market.call('item', params)
     .map(
       (listings: Array<Listing>) => {
+        console.log(listings);
```
[the remark this answers is not in the document]
Comment (vikas-cis) [reply src/app/core/market/api/listing/listing.service.ts]: yeah it is for debug and yeah we change it in to log :)

Thanks

File: src/app/market/sell/seller-listing/seller-listing.component.ts [hunk 2]
```diff
@@ -64,7 +64,8 @@ export class SellerListingComponent {
   }
 
   async callTemplate(template: Template) {
-    await this.template.post(template, 1).toPromise();
+    console.log(template);
```
[the remark this answers is not in the document]
Comment (vikas-cis) [reply src/app/market/sell/seller-listing/seller-listing.component.ts]: I will change it in to debug log rather than console :)

**Comment by kewde** [comment]
Please rebase and fix the conflicts.

**Comment by vikas-cis** [comment]
@kewde I have just fixed conflicts 

Thanks

File: src/app/market/sell/add-item/add-item.component.ts [hunk 3]
```diff
@@ -372,7 +372,10 @@ export class AddItemComponent implements OnInit, OnDestroy {
           this.log.d(listing);
           this.backToSell();
         },
-        (error) => this.snackbar.open(error),
+        (error) => {
+          this.isInProcess = false;
```
Comment (kewde) [inline src/app/market/sell/add-item/add-item.component.ts]: rm

Comment (kewde) [reply src/app/market/sell/add-item/add-item.component.ts]: complete() isn't called after error, it does not behave like finally() for some odd reason. I wasn't aware.

**Pull request merged**